extensions/amp-brightcove/1.0/component.type.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @externs */
+
+/** @const */
+var BrightcoveDef = {};
+
+/**
+ * @typedef {{
+ *   account: (string|undefined),
+ *   autoplay: (boolean|undefined),
+ *   embed: (string|undefined),
+ *   muted: (boolean|undefined),
+ *   onReady: (function(!JsonObject, string|undefined)|undefined),
+ *   player: (string|undefined),
+ *   playlistId: (string|undefined),
+ *   videoId: (string|undefined),
+ *   updatePlaying: (function(boolean)|undefined),
+ *   urlParams: (JsonObject|undefined),
+ * }}
```

Comment:
nit: did you know you don't actually need to declare a variable? I believe you can create a type purely in the docs (the only part I'm unsure about is assigning to an actual variable, may need to also do this with BrightCoveDef):

```js
/**
* @typedef {{...}} BrightCoveDef.Props
*/
```

Reply:
If I understand correctly, the recommendation is to remove the `var BrightcoveDef = {};` line. 🤯 Had no idea we didn't need the var. Now I'm trying to remember why it was introduced in the first place...

Reply:
Most people don't know we can define the type without a var. That said...none of this is currently going through closure's type checker....so no way for us to know what does and doesn't work

Reply:
Checked manually and this didn't work with Closure! Thanks for updating and sorry for making you run in circles